knowledge-base/grid-customize-hierarchy-expand-column-blazor-grid.md

```diff
@@ -0,0 +1,176 @@
+
+To customize the hierarchy expand/collapse column, follow these steps:
+
+1. Hide the default hierarchy expand column using [custom CSS]({%slug themes-override%}).
+2. Add another [column with a template]({%slug grid-templates-column%}) for expanding and collapsing the detail templates through the [Grid state]({%slug components/grid/features/hierarchy%}#expand-rows-from-code).
+3. Disable the data operations for this column if you have enabled them in the Grid (filtering, sorting, etc.)
+4. Manage the icon type in the custom hierarchy expand column based on the item's expanded state.
+5. Configure the custom hierarchy expand column as needed. For example, managing its order within the `GridColumns` declaration or enabling the `ColumnMenu` to lock the column.
+
+Here is a basic example demonstrating the approach: 
+
+````CSHTML
+<TelerikGrid @ref="@GridRef"
+             Data="@GridData"
+             Pageable="true"
+             Sortable="true"
+             ShowColumnMenu="true"
+             FilterMode="@GridFilterMode.FilterMenu"
+             Class="no-default-hierarchy-cell"
+             Width="600px">
+    <GridSettings>
+        <GridColumnMenuSettings ShowColumnChooser="false"/>
+    </GridSettings>
+    <DetailTemplate>
+        @{
+            var employee = context as MainModel;
+
+            <TelerikGrid Data="employee.Orders" Pageable="true" PageSize="5">
+                <GridColumns>
+                    <GridColumn Field="OrderId"></GridColumn>
+                    <GridColumn Field="DealSize"></GridColumn>
+                </GridColumns>
+            </TelerikGrid>
+        }
+    </DetailTemplate>
+    <GridColumns>       
+        <GridColumn Field="@nameof(MainModel.Id)" Width="60px"></GridColumn>
+        <GridColumn Field="@nameof(MainModel.Name)" ></GridColumn>
+        <GridColumn Field="@nameof(MainModel.Address)"></GridColumn>
+        <GridColumn Field="@nameof(MainModel.Email)"></GridColumn>
+        <GridColumn Width="60px" Filterable="false" Sortable="false">
+            <Template>
+                @{
+                    var dataItem = (MainModel)context;
+                }
+                <TelerikButton Icon="@GetButtonIcon(dataItem)" FillMode="@Telerik.Blazor.ThemeConstants.Button.FillMode.Flat" OnClick="@( () => OnButtonClick(dataItem) )" />
+            </Template>
+        </GridColumn>        
+    </GridColumns>
+</TelerikGrid>
+
+<style>
+    /* shrink hierarchy column - do not hide it completely */
+    .no-default-hierarchy-cell .k-hierarchy-col {
+        width: 0;
+    }
+    /* hide everything from hierarchy column */
+    .no-default-hierarchy-cell .k-hierarchy-cell * {
+        display: none;
+    }
+</style>
+
+@code {
+    private TelerikGrid<MainModel> GridRef { get; set; } = null!;
```

Comment:
A nullable `@ref` is technically more correct. In some scenarios the Grid may not be always rendered, so `null!` will be misleading for both the compiler and the developer. You will also need some null checks in the code, like in https://docs.telerik.com/blazor-ui/components/grid/filter/searchbox#search-from-code

Alternatively, you can use `GridRef!` instead of null checks when it's clear that the Grid reference exists. For example, if an event fires from inside the Grid, obviously the component exists.

```suggestion
    private TelerikGrid<MainModel>? GridRef { get; set; }
```